crypto/openssl/openssl_key_exchange_ecdh.c

```diff
@@ -589,7 +585,7 @@ int32_t crypto_hal_set_peer_random(void *context,
  *  @return 0 on success or -1 on failure.
  */
 int32_t crypto_hal_get_secret(void *context, uint8_t *secret,
-			      uint32_t *secret_length)
+		uint32_t *secret_length)
 {
 	ecdh_context_t *key_ex_data = (ecdh_context_t *)context;
 
```

Comment:
Why context is cleared or freed at the end if there is any error

Reply:
Context is cleared in the single function crypto_hal_kex_close() that is invoked separately as part of fdoprot_TO2Exit(). fdoprot_TO2Exit will get invoked in case of all error scenarios like failure of crypto_hal_get_secret